sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -311,15 +311,23 @@ public final class StorageBlobClient: PipelineClient {
     /// - Parameters:
     ///   - blob: The name of the blob.
     ///   - container: The name of the container.
+    ///   - destinationURL: The URL to a file path on this device.
     ///   - options: A `DownloadBlobOptions` object to control the download operation.
     ///   - completion: A completion handler that receives a `BlobDownloader` object on success.
     public func rawDownload(
         blob: String,
         fromContainer container: String,
+        to destinationURL: URL,
```

Comment:
I'm fine with this--just seemed juxtaposed against the theme of working exclusively with URLs. I could see an instance where someone might want to download static URLs, and thus would prefer to just `download(from: to:)` or (eventually) `copy(from: to:)` but we can always add that later on demand. The extra step of having to parse out a URL to blob and container name seems unnecessary.

Reply:
Agreed about that, if we add `copy()` we will definitely want to make both parameters URLs, or both parameters container/blob name pairs. My assumption (which would be nice to validate) is that customers expect to / are used to referring to the blob name and container name pair, rather than the URL of the blob. So I attempted to make sure this was always what the public API presents to the customer.